packages/console/src/hooks/use-tenants.ts

```diff
@@ -0,0 +1,66 @@
+import { useLogto } from '@logto/react';
+import { type TenantInfo } from '@logto/schemas';
+import { type Optional, trySafe } from '@silverhand/essentials';
+import type ky from 'ky';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
+import { TenantsContext } from '@/contexts/TenantsProvider';
+
+type KyInstance = typeof ky;
+
+const useTenants = (cloudApi: KyInstance) => {
+  const { signIn, getAccessToken } = useLogto();
+  const { tenants, setTenants, currentTenantId, setCurrentTenantId, setIsSettle, navigate } =
+    useContext(TenantsContext);
+  const [error, setError] = useState<Error>();
+
+  const validate = useCallback(
+    async (tenant: TenantInfo) => {
+      const { id, indicator } = tenant;
+      if (await trySafe(getAccessToken(indicator))) {
+        setIsSettle(true);
+      } else {
+        void signIn(new URL(`/${id}/callback`, window.location.origin).toString());
+      }
+    },
+    [getAccessToken, setIsSettle, signIn]
+  );
+
+  const loadTenants = useCallback(async () => {
+    try {
+      const availableTenants = await cloudApi.get('/api/tenants').json<TenantInfo[]>();
+      setTenants(availableTenants);
+    } catch (error: unknown) {
+      setError(error instanceof Error ? error : new Error(String(error)));
+    }
+  }, [cloudApi, setTenants]);
+
+  useEffect(() => {
+    if (!tenants) {
+      void loadTenants();
+    }
+  }, [loadTenants, tenants]);
+
+  const currentTenant: Optional<TenantInfo> = useMemo(() => {
+    return tenants?.find(({ id }) => id === currentTenantId);
+  }, [currentTenantId, tenants]);
+
+  useEffect(() => {
+    if (currentTenant) {
+      void validate(currentTenant);
+    }
+  }, [currentTenant, navigate, setCurrentTenantId, tenants, validate]);
```

Comment:
The dependency list seems redundant? Try to removing some unnecessary ones

Reply:
This dependency array needs update

Reply:
updated